packages/eslint-plugin/src/rules/array-type.ts

```diff
@@ -199,13 +202,20 @@ export default createRule<Options, MessageIds>({
           node.typeName.type !== AST_NODE_TYPES.Identifier ||
           !(
             node.typeName.name === 'Array' ||
-            node.typeName.name === 'ReadonlyArray'
+            node.typeName.name === 'ReadonlyArray' ||
+            node.typeName.name === 'Readonly'
           )
         ) {
           return;
         }
 
-        const isReadonlyArrayType = node.typeName.name === 'ReadonlyArray';
+        const isReadonlyWithGenericArrayType =
+          node.typeName.name === 'Readonly' &&
+          node.typeArguments?.params[0].type === AST_NODE_TYPES.TSArrayType;
```

Comment:
[Bug] If the type name is `Readonly` and first type argument is not an array, we should return from the function. Because otherwise any `Readonly<...>` type will be reported.

```ts
const x: Readonly<string> = 'a'
```

[playground](https://deploy-preview-8752--typescript-eslint.netlify.app/play/#ts=5.4.2&fileType=.ts&code=MYewdgzgLgBAHgLhgJQKYEMAm4A2BPAHmgCcBLMAcwD4YBeGAcnQaA&eslintrc=N4KABGBEBOCuA2BTAzpAXGUEKQAIBcBPABxQGNoBLY-AWhXkoDt8B6AQ2mncNqNPRREXAPbRI4MAF8QUoA&tsconfig=&tokens=false)